lib/domain/provider/pollution/pollution_provider_model_state.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:smoge/data/serialization/pollution_data.dart';
+import 'package:smoge/data/serialization/pollution_quality_index.dart';
+import 'package:smoge/data/serialization/pollution_sensor.dart';
+import 'package:smoge/data/serialization/pollution_station.dart';
+import 'package:smoge/domain/provider/provider_model_async_result.dart';
+
+class PollutionProviderModelState {
+  ProviderModelAsyncResult<PollutionStation, Exception> firstStation =
+      ProviderModelAsyncResult<PollutionStation, Exception>();
```

Comment:
Ta klasa mnie trochę gryzie w oczy. Nie chodzi o kod tylko o styl. Jest bardzo WhenPlacowy xD
Pamiętam że na początku ustaliliście sobie że piszecie wszystkie typy wszędzie. Od tego czasu inferencja typów w Darcie mocno się poprawiła, do tego stopnia że na stronie Darta piszą:
**AVOID type annotating initialized local variables**
https://dart.dev/guides/language/effective-dart/design#avoid-type-annotating-initialized-local-variables

Jeżeli mamy tym blog postem edukować to jestem za tym żeby edukować zgodnie z tym co devowie Darta mówią 😇 

Moja sugestia:
```suggestion
  final firstStation = ProviderModelAsyncResult<PollutionStation, Exception>();
```

I dodać do projektu to:
https://dart.dev/guides/language/analysis-options#enabling-linter-rules

Reply:
wprowadzilem analyzera i mnooostwo warningow sie pojawilo